### PR TITLE
feat(docker): Add Docker Registry support

### PR DESCRIPTION
**Pull request by sergio-quintero**
After `kork-credentials` was implemented to `clouddriver-docker` accounts https://github.com/spinnaker/clouddriver/pull/5593, we are adding support for external accounts.

This changes break the `release-0.1.x` and `release-0.2.x` and are not compatible with Spinnaker 1.27 since we are working with clouddriver master branch. Once Spinnaker `1.28` gets released, these changes can be merged to a `release-0.3.x` branch.

### CODE REVIEW

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,5 +1,5 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-6.6.1-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-6.8.1-bin.zip
```
Comment (sergio-quintero) [inline gradle/wrapper/gradle-wrapper.properties:3]: Updating Gradle to the same version used by Clouddriver https://github.com/spinnaker/clouddriver/blob/master/gradle/wrapper/gradle-wrapper.properties#L3

**Comment by sergio-quintero** [comment]
This was tested locally but we can provide evidence if needed.

File: common/common.gradle [hunk 2]
```diff
@@ -10,12 +10,15 @@ sourceCompatibility = 1.11
 targetCompatibility = 1.11
 
 dependencies {
-    compileOnly(group: 'com.google.guava', name: 'guava', version: '28.2-jre')
-    compileOnly(group: 'org.springframework.boot', name: 'spring-boot', version: '2.2.5.RELEASE')
-    compileOnly(group: 'com.fasterxml.jackson.core', name: 'jackson-annotations', version: '2.11.2')
-    compileOnly(group: 'javax.annotation', name: 'javax.annotation-api', version: '1.3.2')
+    compileOnly(enforcedPlatform("io.spinnaker.clouddriver:clouddriver-bom:${clouddriverVersion}"))
```
Comment (german-muzquiz) [inline common/common.gradle:13]: What's the purpose of this dependency?

**Review by german-muzquiz — COMMENTED** [review]
I only have a couple of questions. Good work with the integration tests!

File: clouddriver/src/main/java/io/armory/plugin/eap/EAPPlugin.java [hunk 3]
```diff
@@ -52,7 +60,25 @@ public void stop() {
     @Override
     public void registerBeanDefinitions(BeanDefinitionRegistry registry) {
         super.registerBeanDefinitions(registry);
-        registry.getBeanDefinition("cloudDriverConfig").
```
[the remark this answers is not in the document]
Comment (nemesisOsorio) [reply clouddriver/src/main/java/io/armory/plugin/eap/EAPPlugin.java:55]: yep, there were circular dependencies for spring beans, we tried to use only `cloudDriverConfig` but we need to change a lot of Clouddriver beans in OSS, and we prefer to not touch multiple providers in Clouddriver

File: common/common.gradle (hunk 2 above)
Comment (nemesisOsorio) [reply common/common.gradle:13]: I believe is to use transitive dependency versions from Clouddriver
same as `clouddriver/clouddriver.gradle` https://github.com/armory-plugins/external-accounts/blob/a7f7f96e448a6b44bb046a97cdcd47b38c4bd38f/clouddriver/clouddriver.gradle#L25

**Review by german-muzquiz — APPROVED** [review]

**Pull request merged**